src/lib/mina_generators/zkapp_command_generators.ml

```diff
@@ -1483,3 +1524,22 @@ let gen_list_of_zkapp_command_from ?failure ?max_account_updates
     else return (List.rev acc)
   in
   go length []
+
+let gen_zkapp_commands_with_limited_keys ~keymap ?account_state_tbl ~ledger
+    ?protocol_state_view ?vk ?num_account_updates
+    ~(fee_payer_keypair : Signature_lib.Keypair.t) () =
+  let open Quickcheck.Generator.Let_syntax in
+  let pks = Signature_lib.Public_key.Compressed.Map.keys keymap in
+  let%bind balancing_account_id =
+    let%map index = Int.gen_uniform_incl 0 (List.length pks - 1) in
+    Account_id.create (List.nth_exn pks index) Token_id.default
+  in
+  let num_account_updates =
+    Option.map num_account_updates ~f:(fun n -> `Fixed n)
+  in
+  let limited_zkapp_accounts =
```

Comment:
This variable is called `limited_zkapp_accounts` but it just treat all pk in `keymap` as zkapp accounts. Is this intentional?

Reply:
This doesn't seem right unless all the keys in the keymap are for zkapp accounts.

Reply:
They are all zkapp accounts, indeed. There is a fee payer key in there tagged as `Fee_payer`